// together-ai/index.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var _Together_instances, _a, _Together_baseURLOverridden;
import * as Core from "./core.mjs";
import * as Errors from "./error.mjs";
import * as Uploads from "./uploads.mjs";
import * as API from "./resources/index.mjs";
import { Batches, } from "./resources/batches.mjs";
import { Completions, } from "./resources/completions.mjs";
import { Embeddings } from "./resources/embeddings.mjs";
import { Endpoints, } from "./resources/endpoints.mjs";
import { Files, } from "./resources/files.mjs";
import { FineTuneResource, } from "./resources/fine-tune.mjs";
import { Hardware } from "./resources/hardware.mjs";
import { Images } from "./resources/images.mjs";
import { Jobs } from "./resources/jobs.mjs";
import { Models } from "./resources/models.mjs";
import { Audio, } from "./resources/audio/audio.mjs";
import { Chat } from "./resources/chat/chat.mjs";
import { CodeInterpreter, } from "./resources/code-interpreter/code-interpreter.mjs";
/**
 * API Client for interfacing with the Together API.
 */
export class Together extends Core.APIClient {
    /**
     * API Client for interfacing with the Together API.
     *
     * @param {string | undefined} [opts.apiKey=process.env['TOGETHER_API_KEY'] ?? undefined]
     * @param {string} [opts.baseURL=process.env['TOGETHER_BASE_URL'] ?? https://api.together.xyz/v1] - Override the default base URL for the API.
     * @param {number} [opts.timeout=1 minute] - The maximum amount of time (in milliseconds) the client will wait for a response before timing out.
     * @param {number} [opts.httpAgent] - An HTTP agent used to manage HTTP(s) connections.
     * @param {Core.Fetch} [opts.fetch] - Specify a custom `fetch` function implementation.
     * @param {number} [opts.maxRetries=5] - The maximum number of times the client will retry a request.
     * @param {Core.Headers} opts.defaultHeaders - Default headers to include with every request to the API.
     * @param {Core.DefaultQuery} opts.defaultQuery - Default query parameters to include with every request to the API.
     */
    constructor({ baseURL = Core.readEnv('TOGETHER_BASE_URL'), apiKey = Core.readEnv('TOGETHER_API_KEY'), ...opts } = {}) {
        if (apiKey === undefined) {
            throw new Errors.TogetherError("The TOGETHER_API_KEY environment variable is missing or empty; either provide it, or instantiate the Together client with an apiKey option, like new Together({ apiKey: 'My API Key' }).");
        }
        const options = {
            apiKey,
            ...opts,
            baseURL: baseURL || `https://api.together.xyz/v1`,
        };
        super({
            baseURL: options.baseURL,
            baseURLOverridden: baseURL ? baseURL !== 'https://api.together.xyz/v1' : false,
            timeout: options.timeout ?? 60000 /* 1 minute */,
            httpAgent: options.httpAgent,
            maxRetries: options.maxRetries,
            fetch: options.fetch,
        });
        _Together_instances.add(this);
        this.chat = new API.Chat(this);
        this.completions = new API.Completions(this);
        this.embeddings = new API.Embeddings(this);
        this.files = new API.Files(this);
        this.fineTune = new API.FineTuneResource(this);
        this.codeInterpreter = new API.CodeInterpreter(this);
        this.images = new API.Images(this);
        this.audio = new API.Audio(this);
        this.models = new API.Models(this);
        this.jobs = new API.Jobs(this);
        this.endpoints = new API.Endpoints(this);
        this.hardware = new API.Hardware(this);
        this.batches = new API.Batches(this);
        this._options = options;
        this.apiKey = apiKey;
    }
    /**
     * Query a reranker model
     *
     * @example
     * ```ts
     * const response = await client.rerank({
     *   documents: [
     *     { title: 'bar', text: 'bar' },
     *     { title: 'bar', text: 'bar' },
     *     { title: 'bar', text: 'bar' },
     *     { title: 'bar', text: 'bar' },
     *   ],
     *   model: 'Salesforce/Llama-Rank-V1',
     *   query: 'What animals can I find near Peru?',
     * });
     * ```
     */
    rerank(body, options) {
        return this.post('/rerank', { body, ...options });
    }
    defaultQuery() {
        return this._options.defaultQuery;
    }
    defaultHeaders(opts) {
        return {
            ...super.defaultHeaders(opts),
            ...this._options.defaultHeaders,
        };
    }
    authHeaders(opts) {
        return { Authorization: `Bearer ${this.apiKey}` };
    }
}
_a = Together, _Together_instances = new WeakSet(), _Together_baseURLOverridden = function _Together_baseURLOverridden() {
    return this.baseURL !== 'https://api.together.xyz/v1';
};
Together.Together = _a;
Together.DEFAULT_TIMEOUT = 60000; // 1 minute
Together.TogetherError = Errors.TogetherError;
Together.APIError = Errors.APIError;
Together.APIConnectionError = Errors.APIConnectionError;
Together.APIConnectionTimeoutError = Errors.APIConnectionTimeoutError;
Together.APIUserAbortError = Errors.APIUserAbortError;
Together.NotFoundError = Errors.NotFoundError;
Together.ConflictError = Errors.ConflictError;
Together.RateLimitError = Errors.RateLimitError;
Together.BadRequestError = Errors.BadRequestError;
Together.AuthenticationError = Errors.AuthenticationError;
Together.InternalServerError = Errors.InternalServerError;
Together.PermissionDeniedError = Errors.PermissionDeniedError;
Together.UnprocessableEntityError = Errors.UnprocessableEntityError;
Together.toFile = Uploads.toFile;
Together.fileFromPath = Uploads.fileFromPath;
Together.Chat = Chat;
Together.Completions = Completions;
Together.Embeddings = Embeddings;
Together.Files = Files;
Together.FineTuneResource = FineTuneResource;
Together.CodeInterpreter = CodeInterpreter;
Together.Images = Images;
Together.Audio = Audio;
Together.Models = Models;
Together.Jobs = Jobs;
Together.Endpoints = Endpoints;
Together.Hardware = Hardware;
Together.Batches = Batches;
export { toFile, fileFromPath } from "./uploads.mjs";
export { TogetherError, APIError, APIConnectionError, APIConnectionTimeoutError, APIUserAbortError, NotFoundError, ConflictError, RateLimitError, BadRequestError, AuthenticationError, InternalServerError, PermissionDeniedError, UnprocessableEntityError, } from "./error.mjs";
export default Together;
