// together-ai/resources/endpoints.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Endpoints = void 0;
const resource_1 = require("../resource.js");
const core_1 = require("../core.js");
class Endpoints extends resource_1.APIResource {
    /**
     * Creates a new dedicated endpoint for serving models. The endpoint will
     * automatically start after creation. You can deploy any supported model on
     * hardware configurations that meet the model's requirements.
     *
     * @example
     * ```ts
     * const endpoint = await client.endpoints.create({
     *   autoscaling: { max_replicas: 5, min_replicas: 2 },
     *   hardware: '1x_nvidia_a100_80gb_sxm',
     *   model: 'meta-llama/Llama-3-8b-chat-hf',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/endpoints', { body, ...options });
    }
    /**
     * Retrieves details about a specific endpoint, including its current state,
     * configuration, and scaling settings.
     *
     * @example
     * ```ts
     * const endpoint = await client.endpoints.retrieve(
     *   'endpoint-d23901de-ef8f-44bf-b3e7-de9c1ca8f2d7',
     * );
     * ```
     */
    retrieve(endpointId, options) {
        return this._client.get(`/endpoints/${endpointId}`, options);
    }
    /**
     * Updates an existing endpoint's configuration. You can modify the display name,
     * autoscaling settings, or change the endpoint's state (start/stop).
     *
     * @example
     * ```ts
     * const endpoint = await client.endpoints.update(
     *   'endpoint-d23901de-ef8f-44bf-b3e7-de9c1ca8f2d7',
     * );
     * ```
     */
    update(endpointId, body, options) {
        return this._client.patch(`/endpoints/${endpointId}`, { body, ...options });
    }
    list(query = {}, options) {
        if ((0, core_1.isRequestOptions)(query)) {
            return this.list({}, query);
        }
        return this._client.get('/endpoints', { query, ...options });
    }
    /**
     * Permanently deletes an endpoint. This action cannot be undone.
     *
     * @example
     * ```ts
     * await client.endpoints.delete(
     *   'endpoint-d23901de-ef8f-44bf-b3e7-de9c1ca8f2d7',
     * );
     * ```
     */
    delete(endpointId, options) {
        return this._client.delete(`/endpoints/${endpointId}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
}
exports.Endpoints = Endpoints;
